src/lib/UniversalDiscovery/ConfigResolver.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\UniversalDiscovery;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use EzSystems\EzPlatformAdminUi\UniversalDiscovery\Event\ConfigResolveEvent;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+class ConfigResolver
+{
+    /** @var \Symfony\Component\EventDispatcher\EventDispatcher */
+    protected $eventDispatcher;
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    protected $configResolver;
+
+    /** @var array */
+    protected $udwConfiguration;
+
+    /**
+     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
+     * @param \Symfony\Component\EventDispatcher\EventDispatcher $eventDispatcher
+     * @param array $udwConfiguration
+     */
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        EventDispatcher $eventDispatcher,
+        array $udwConfiguration
+    ) {
+        $this->configResolver = $configResolver;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->udwConfiguration = $udwConfiguration;
+    }
+
+    /**
+     * @param string $configName
+     * @param array $context
+     *
+     * @return array
+     */
+    public function getConfig(string $configName, array $context = []): array
+    {
+        $config = $this->udwConfiguration[$configName] ?? [];
+        $defaults = $this->udwConfiguration['default'] ?? [];
+
+        $config = $this->mergeConfiguration($defaults, $config);
+
+        $configResolveEvent = new ConfigResolveEvent();
+
+        $configResolveEvent->setConfigName($configName);
```

Comment:
Meh, better create a constructor in `ConfigResolveEvent` and initialize properties there 😃